packages/runtime/src/routeGeneration/templateHelpers.ts

```diff
@@ -6,14 +6,23 @@ import { Tsoa } from '../metadataGeneration/tsoa';
 import ValidatorKey = Tsoa.ValidatorKey;
 
 // for backwards compatibility with custom templates
-export function ValidateParam(property: TsoaRoute.PropertySchema, value: any, generatedModels: TsoaRoute.Models, name = '', fieldErrors: FieldErrors, parent = '', swaggerConfig: AdditionalProps) {
-  return new ValidationService(generatedModels).ValidateParam(property, value, name, fieldErrors, parent, swaggerConfig);
+export function ValidateParam(
+  property: TsoaRoute.PropertySchema,
+  value: any,
+  generatedModels: TsoaRoute.Models,
+  name = '',
+  fieldErrors: FieldErrors,
+  isBodyParam: boolean,
+  parent = '',
+  swaggerConfig: AdditionalProps,
+) {
+  return new ValidationService(generatedModels).ValidateParam(property, value, name, fieldErrors, isBodyParam, parent, swaggerConfig);
 }
 
 export class ValidationService {
   constructor(private readonly models: TsoaRoute.Models) {}
```

Comment:
I am not sure, do we ever need to swap the minimalSwaggerConfig: AdditionalProps AND the isBodyParam? maybe we can init the ValidationService with them and avoid carrying the 2 new params around all the time.

Reply:
We can probably move the `minimalSwaggerConfig` to be a property on the ValidationService because it won't change at runtime. The `isBodyParam` parameter though is tied closely to the value being validated. Thats why I think we need to keep this one as a parameter which is carried through. WDYT?

Reply:
yeah, let's move minimalSwaggerConfig to the constructor (deep optional).
I just checked and yes, since we reuse the validation service, keep the isBody as arg to validate.

Reply:
I was in the midst of doing this change when I found these pieces of code:
- https://github.com/friedow/tsoa/blob/feat/configure-implicit-body-coercion/packages/runtime/src/routeGeneration/templateHelpers.ts#L603-L606
- https://github.com/friedow/tsoa/blob/feat/configure-implicit-body-coercion/packages/runtime/src/routeGeneration/templateHelpers.ts#L572-L575

This changes the swaggerConfig of validation at runtime. If I move this config to be a property on the `ValidationService` this would unfortunately become a breaking change :/.

Reply:
Theoretically, you could just run these valudations not on this, but on a new subValidationService.

Reply:
I like this. Will refactor it like this :).

Reply:
Done ;)